python/ray/experimental/sgd/sgd.py

```diff
@@ -177,6 +180,33 @@ def warmup(self):
         ray.get([w.warmup.remote() for w in self.workers])
         logger.info("Warmup complete")
 
+    def save_checkpoint(self, path):
+        param_oids = [worker.get_params.remote() for worker in self.workers]
+        model_count = 0
+        while param_oids:
+            ready_oids, param_oids = ray.wait(param_oids)
+            params_list = ray.get(ready_oids)[0]
+            for params in params_list:
+                filename = os.path.join(path, "model_%d.pkl" % model_count)
+                with open(filename, "wb") as f:
+                    pickle.dump(params, f)
+                model_count += 1
+
+            ray.internal.free(ready_oids)
+
+    def restore_checkpoint(self, path):
```

Comment:
Use this: https://github.com/ray-project/ray/blob/master/python/ray/experimental/sgd/mnist_example.py#L116